brian2/core/variables.py

```diff
@@ -274,6 +275,9 @@ def __repr__(self):
                                   constant=repr(self.constant),
                                   read_only=repr(self.read_only))
 
+    _state_tuple = property(lambda self: (self.dim, self.dtype, self.scalar,
+                                          self.constant, self.read_only,
+                                          self.dynamic, self.name))
```

Comment:
Ah, sorry, overlooked this comment... Hmm, I guess we could use `__dict__` to get all attributes and do something along these lines. I'll give it a go!